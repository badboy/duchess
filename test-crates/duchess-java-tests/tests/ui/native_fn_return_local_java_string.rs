use duchess::{java, prelude::*};

duchess::java_package! {
    package native_greeting;

    public class native_greeting.Native {
        public native_greeting.Native();
        public java.lang.String greet(java.lang.String);
        native java.lang.String baseGreeting(java.lang.String);
    }
}

#[duchess::java_function(native_greeting.Native::baseGreeting)]
fn base_greeting<'n>(
    _this: &native_greeting::Native,
    name: &'n java::lang::String,
) -> duchess::GlobalResult<duchess::Local<'n, java::lang::String>> {
    name.execute() //~ ERROR: mismatched types
}

fn main() -> duchess::GlobalResult<()> {
    duchess::Jvm::builder()
        .link(base_greeting::java_fn())
        .try_launch()?;

    let n: String = native_greeting::Native::new()
        .greet("Ferris")
        .assert_not_null()
        .to_rust()
        .unwrap();

    assert_eq!(n, "Ferris, from Java");

    Ok(())
}
