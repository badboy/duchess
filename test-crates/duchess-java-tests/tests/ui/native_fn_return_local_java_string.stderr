error[E0308]: mismatched types
  --> $DIR/native_fn_return_local_java_string.rs:18:5
   |
17 | ) -> duchess::Result<duchess::Local<'n, java::lang::String>> {
   |      ------------------------------------------------------------- expected `Result<Local<'n, duchess::java::lang::String>, duchess::Error<duchess::Java<Throwable>>>` because of return type
18 |     name.execute()
   |     ^^^^^^^^^^^^^^ expected `Result<Local<'_, String>, ...>`, found `Result<&String, Error<Java<...>>>`
   |
   = note: expected enum `Result<Local<'n, duchess::java::lang::String, >, _>`
              found enum `Result<&duchess::java::lang::String, _>`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0308`.
